internal/build/imgsrc/dockerfile_builder.go

```diff
@@ -414,7 +415,15 @@ func runBuildKitBuild(ctx context.Context, docker *dockerclient.Client, opts Ima
 	return res.ExporterResponse[exptypes.ExporterImageDigestKey], nil
 }
 
-func pushToFly(ctx context.Context, docker *dockerclient.Client, streams *iostreams.IOStreams, tag string) error {
+func pushToFly(ctx context.Context, docker *dockerclient.Client, streams *iostreams.IOStreams, tag string) (err error) {
+	ctx, span := tracing.GetTracer().Start(ctx, "push_image_to_fly")
```

Comment:
Could this just be "push_image"? It defaults to registry.fly.io but technically it could be anything.